BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift

```diff
@@ -261,14 +290,41 @@ public final class ConvivaAnalytics: NSObject {
         self.isBumper = false
         logger.debugLog(message: "Tracking resumed.")
     }
+
+    /// Attaches a `Player` instance to the Conviva Analytics object.
+    /// We suggest to call this method as soon as the `Player` instance is initialized to not mis any tracking.
+    ///
+    /// Has no effect if there is already an `Player` instance set. Use the `ConvivaAnalytics.init` without `player`
+    /// if you plan to attach an `Player` instance later in the life-cycle.
+    public func attach(player: Player) {
```

Comment:
This is new API

Reply:
So, what do you think about naming this method `attachPlayer` to align with the rest of the methods?

Reply:
At least in the web integration, I think it makes sense to return a boolean to indicate if the player instance has been attached because there are quite some checks in the method:

- https://github.com/bitmovin/bitmovin-player-web-analytics-conviva/pull/115/files#diff-ef4a4ddb88b124b585d817d8a3b32d8cd69a320238398b41f39fd7b6906e6b1bR103
- https://github.com/bitmovin/bitmovin-player-web-analytics-conviva/pull/115/files#diff-e85097e0eaba7b0240d78adc883916dd4ceecf3bb589d15f10ccd87e0ebd9cc2R180

Reply:
it is called `attachPlayer` 🤔 `attach(player:)` is idiomatic Swift and reads as `.attach(player: player)` when using. We prefer platform/language specifics over 100% API parity on mobile.

And for the second:
We have the same checks but it's not common to return a boolean indicating success or failure. The Swift way would be to throw errors but we agreed (via slack) on gracefully handling and only do logging.

Reply:
> but we agreed (via slack) on gracefully handling and only do logging

It means there is no way to know if the player has been attached programmatically. We don't throw errors and we do not return anything. Might it be a problem?

Reply:
>  is idiomatic Swift and reads as `.attach(player: player)`

In TypeScript it's not always clear what `attach` means, e.g. in this outline:

<img width="431" alt="image" src="https://github.com/user-attachments/assets/8e7c6bf8-5a03-4406-b0f1-949dbf472e99">

Maybe then I should keep it at `attachPlayer` in the web implementation. 🤔

Reply:
> It means there is no way to know if the player has been attached programmatically. We don't throw errors and we do not return anything. Might it be a problem?

I don't think it will be a problem 🤔 After calling `attachPlayer` there will be for sure a `Player` instance attached. Either there was already one before or the current one was `attached`.

Reply:
> Maybe then I should keep it as attachPlayer in the web implementation. 🤔

Yes, on Web and Android it should still be called `attachPlayer()`. Just Swift is the outlier here.